specs/AppLifecycle/Restart/restartApi.markdown

```diff
@@ -64,135 +69,177 @@ There are 2 related Win32 APIs that focus on recovery prior to restart:
-<td><p>Equivalent to CoreApplication.RequestRestartAsync, except that it is <em>not</em> async.
-<ul>
-<li>If the restart fails, but the user subsequently launches the app manually, the app will launch normally and no restart arguments will be passed. </li>
-<li>If the app has any in-process background tasks running when it calls this API, those tasks will be cancelled in the normal way. Out-of-process background tasks will not be affected. </li>
-</ul>
- The restarted instance will be activated with Windows.ApplicationModel.Activation.LaunchActivatedEventArgs.</p>
- <ul>
-<li>When the app is restarted, LaunchActivatedEventArgs.PreviousExecutionState will have the value Terminated so that the app can distinguish between a resume and a restart. </li>
-</ul>
-<p>If the request to restart is successful, the application will be restarted immediately. The user does not have to do anything else/handle a return value in any way.
+### New API
+
+```c#
+static  AppRestartFailureReason Restart(String arguments)
+```
+
+```c#
+namespace Microsoft.Windows.AppLifecycle
+{
+    enum ExtendedActivationKind
+    {
+        Launch = 0,
+        Search,
+        ShareTarget,
+        File,
+        Protocol,
+        FileOpenPicker,
+        FileSavePicker,
+        CachedFileUpdater,
+        ContactPicker,
+        Device,
+        PrintTaskSettings,
+        CameraSettings,
+        RestrictedLaunch,
+        AppointmentsProvider,
+        Contact,
+        LockScreenCall,
+        VoiceCommand,
+        LockScreen,
+        PickerReturned = 1000,
+        WalletAction,
+        PickFileContinuation,
+        PickSaveFileContinuation,
+        PickFolderContinuation,
+        WebAuthenticationBrokerContinuation,
+        WebAccountProvider,
+        ComponentUI,
+        ProtocolForResults = 1009,
+        ToastNotification,
+        Print3DWorkflow,
+        DialReceiver,
+        DevicePairing,
+        UserDataAccountsProvider,
+        FilePickerExperience,
+        LockScreenComponent,
+        ContactPanel,
+        PrintWorkflowForegroundTask,
+        GameUIProvider,
+        StartupTask,
+        CommandLineLaunch,
+        BarcodeScannerProvider,
+        PrintSupportJobUI,
+        PrintSupportSettingsUI,
+        PhoneCallActivation,
+        VpnForeground,
+        // NOTE: Values below 5000 are designated for the platform.  The above list is kept in sync with
+        // Windows.ApplicationModel.Activation.ActivationKind.
+
+        Push = 5000,
+    };
+
+    runtimeclass AppActivationArguments
+    {
+        ExtendedActivationKind Kind { get; };
+        IInspectable Data{ get; };
+    };
+
+    runtimeclass AppInstance
+    {
+        static AppInstance GetCurrent();
+        static Windows.Foundation.Collections.IVector<AppInstance> GetInstances();
+        static AppInstance FindOrRegisterForKey(String key);
+        static Windows.ApplicationModel.Core.AppRestartFailureReason Restart(String arguments);
```

Comment:
Consider putting a "this is the new one!" marker here. Also since this is a new API in 1.1, you'll need to add the proper contract information here, like:

```idl
[contractversion(1.1)] apicontract AppLifecycleContract{}

[contract(AppLifecycleContract, 1)]
runtimeclass AppInstance
{

    // On the new thing
    [contract(AppLifecycleContract, 1.1)]
    {
        static Windows.ApplicationModel.Core.AppRestartFailureReason Restart(String arguments);
    }
}
```

Reply:
Added a marker for this.

On the note on contract: Is there a template for how/where the contract information belongs? Do I just drop it into the IDL as you've commented above?